### PR TITLE
Arabic translation 

### PR DESCRIPTION
**Pull request by Samreen998**
Adding i18n support for Time Ago Formatter

### CODE REVIEW

File: src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java [hunk 1]
```diff
@@ -24,111 +23,132 @@ public String getVersion() {
...
-                    return Math.abs(dateDiff.getMonths()) + " month(s) " + Math.abs(dateDiff.getDays()) + " day(s)";
+                    return Math.abs(dateDiff.getMonths()) + getMonth() + Math.abs(dateDiff.getDays()) + getDay();
                 } else {
-                    return Math.abs(dateDiff.getDays()) + " day(s)";
+                    return Math.abs(dateDiff.getDays()) + getDay();
                 }
-                
+
+            } catch (Exception e) {
+                e.printStackTrace();
```
Comment (hugoatjoget) [inline src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java]: please use back LogUtil.error

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (hugoatjoget) [inline .vscode/settings.json]: please do not include IDE specific files such as .vscode/settings.json

File: src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java [hunk 3]
```diff
@@ -24,111 +23,132 @@ public String getVersion() {
     }
 
     public String getDescription() {
-        //support i18n
-        return AppPluginUtil.getMessage("org.joget.marketplace.TimeAgoDatalistFormatter.pluginDesc", getClassName(), MESSAGE_PATH);
+        // support i18n
+        return AppPluginUtil.getMessage("org.joget.marketplace.TimeAgoDatalistFormatter.pluginDesc", getClassName(),
+                MESSAGE_PATH);
```
Comment (hugoatjoget) [inline src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java]: please keep it to single line. do not need 2 lines.

File: src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java [hunk 4]
```diff
@@ -24,111 +23,132 @@ public String getVersion() {
...
+    }
+
+    public String getYear() {
+        // support i18n
+        return " " + AppPluginUtil.getMessage("org.joget.marketplace.TimeAgoDatalistFormatter.year(s)", getClassName(),
```
Comment (hugoatjoget) [inline src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java]: please keep it to single line. do not need 2 lines.

File: src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java [hunk 5]
```diff
@@ -24,111 +23,132 @@ public String getVersion() {
...
+                MESSAGE_PATH);
+    }
+
+    public String getMonth() {
+        // support i18n
+        return " " + AppPluginUtil.getMessage("org.joget.marketplace.TimeAgoDatalistFormatter.month(s)", getClassName(),
```
Comment (hugoatjoget) [inline src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java]: please keep it to single line. do not need 2 lines.

File: src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java [hunk 6]
```diff
@@ -24,111 +23,132 @@ public String getVersion() {
...
+    public String getDay() {
+        // support i18n
+        return " " + AppPluginUtil.getMessage("org.joget.marketplace.TimeAgoDatalistFormatter.day(s)", getClassName(),
+                MESSAGE_PATH);
     }
 
     @Override
     public String format(DataList dataList, DataListColumn column, Object row, Object value) {
-        
+
```
Comment (hugoatjoget) [inline src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java:47]: why the extra empty line?

File: src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java [hunk 7]
```diff
@@ -24,111 +23,132 @@ public String getVersion() {
...
         AppDefinition appDef = AppUtil.getCurrentAppDefinition();
         String result = (String) value;
+
         String duration = getPropertyString("duration");
-        
-        try {
```
Comment (hugoatjoget) [inline src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java:38]: please retain the try block. do not remove it.

File: src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java (hunk 1 above)
Comment (hugoatjoget) [inline src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java]: please use LogUtil.error. do not use e.printStackTrace.

**Review by hugoatjoget — CHANGES_REQUESTED** [review]
please check out my comments and make the necessary changes. many thanks!

**Comment by Samreen998** [comment]
I did the changes requested

File: src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java [hunk 8]
```diff
@@ -24,103 +23,118 @@ public String getVersion() {
...
-                    return Math.abs(dateDiff.getMonths()) + " month(s) " + Math.abs(dateDiff.getDays()) + " day(s)";
+                    return Math.abs(dateDiff.getMonths()) + getMonth() + Math.abs(dateDiff.getDays()) + getDay();
                 } else {
-                    return Math.abs(dateDiff.getDays()) + " day(s)";
+                    return Math.abs(dateDiff.getDays()) + getDay();
                 }
-                
+
+            } catch (Exception e) {
+                LogUtil.error("org.sample.TimeAgoDatalistFormatter", e, "Error!!");
```
Comment (hugoatjoget) [inline src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java]: no need to change this line.

pls revert to 

LogUtil.error(TimeAgoDatalistFormatter.class.getName(), e, "Not able to compute duration");

File: src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java [hunk 9]
```diff
@@ -24,103 +23,118 @@ public String getVersion() {
...
+                    return Math.abs(dateDiff.getDays()) + getDay();
                 }
 
-            } else if (duration.equals("anotherDate")) {
-                if(result.isEmpty()){
-                    return result;
-                }
-                LocalDate date = LocalDate.parse(result); //Date
+            } catch (Exception e) {
+               LogUtil.error("org.sample.TimeAgoDatalistFormatter", e, "Error!!");
```
Comment (hugoatjoget) [inline src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java:74]: org.sample.TimeAgo is not correct.

File: src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java [hunk 10]
```diff
@@ -24,103 +23,118 @@ public String getVersion() {
...
-                    return Math.abs(dateDiff.getMonths()) + " month(s) " + Math.abs(dateDiff.getDays()) + " day(s)";
+                    return Math.abs(dateDiff.getMonths()) + getMonth() + Math.abs(dateDiff.getDays()) + getDay();
                 } else {
-                    return Math.abs(dateDiff.getDays()) + " day(s)";
+                    return Math.abs(dateDiff.getDays()) + getDay();
                 }
 
-            } else if (duration.equals("twoDates")) {
+            } catch (Exception e) {
+               LogUtil.error("org.sample.TimeAgoDatalistFormatter", e, "Error!!");
```
Comment (hugoatjoget) [inline src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java:100]: same here. pls review. thanks.

File: src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java (hunk 9 above)
Comment (Samreen998) [reply src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java:74]: I have a question here Should it be the same as this one
 LogUtil.error(TimeAgoDatalistFormatter.class.getName(), e, "Not able to compute duration"); 
?

Comment (hugoatjoget) [reply src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java:74]: yup... yes please.

**Pull request merged**